plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,85 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Docker struct {
+	ID   string `toml:"id"`
+	Path string `toml:"path"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if d.Path == "" {
+		// setting the default directory for Docker Secrets
+		// if no explicit path mentioned in configuration
+		d.Path = "/run/secrets"
+	}
+	if _, err := os.Stat(d.Path); os.IsNotExist(err) {
+		// if there is no /run/secrets directory for default Path value
+		// this implies that there are no secrets.
+		// Or for any explicit path definitions for that matter.
+		return fmt.Errorf("directory %s does not exist", d.Path)
+	}
+	return nil
+}
+
+func (d *Docker) Get(key string) ([]byte, error) {
+	secretFile := filepath.Join(d.Path, key)
+	value, err := os.ReadFile(secretFile)
```

Comment:
Can we please make sure that the resulting file is still under the given path (`d.Path`)? This is vulnerable to directory traversals if the user manages to craft a key containing `../` elements. So maybe something along the lines of

```suggestion
	secretFile, err := filepath.Abs(filepath.Join(d.Path, key))
	if err != nil {
		return nil, err
	}
	if filepath.Dir(secretFile) != d.Path {
		return nil, fmt.Errorf("directory traversal detected for key %q", key")
	}

	value, err := os.ReadFile(secretFile)
```